sequencer/src/persistence/sql.rs

```diff
@@ -1346,6 +1349,65 @@ impl SequencerPersistence for Persistence {
     }
 }
 
+#[async_trait]
+impl DhtPersistentStorage for Persistence {
+    /// Save the DHT to the database
+    ///
+    /// # Errors
+    /// - If we fail to serialize the records
+    /// - If we fail to write the serialized records to the DB
+    async fn save(&self, records: Vec<SerializableRecord>) -> anyhow::Result<()> {
+        // Bincode-serialize the records
+        let to_save =
+            bincode::serialize(&records).with_context(|| "failed to serialize records")?;
+
+        // Prepare the statement
+        let stmt = "INSERT OR REPLACE INTO libp2p_dht (id, serialized_records) VALUES (0, $1)";
```

Comment:
I don't think postgres supports` INSERT OR REPLACE`. You can use `ON CONFLICT (id) DO UPDATE`
or there is a `upsert() `on transaction which can do it for you